### PR TITLE
[DSD-4606] Added documentation to change-domain-name-in-a-running-env.md

### PR DESCRIPTION
**Pull request by syedsalman3753**

### CODE REVIEW

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Change domain name in a running env ( v2 )
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:1]: # Change domain name in a running environment (V2)

Check if it needs to be V2 or v2

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+Follow the below steps to rename domain name in a running v2 env:
+1. Update the domain name in groups/all.yml
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:4]: Update the domain name in `groups/all.yml`.

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+2. Ran playbooks/nginx.yml
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:5]: Run `playbooks/nginx.yml`.

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+3. Redeploy config-server, keycloak
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:6]: Re-deploy config-server, keycloak.

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+4. Update  the domain name in the  valid redirect URL & web origins of mosip keycloak-clients
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:7]: Update the domain name in the valid redirect URL & web origins of MOSIP keycloak-clients.

Remove extra spaces

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
+# Change domain name in a running env ( v2 )
+
+Follow the below steps to rename domain name in a running v2 env:
+1. Update the domain name in groups/all.yml
+2. Ran playbooks/nginx.yml
+3. Redeploy config-server, keycloak
+4. Update  the domain name in the  valid redirect URL & web origins of mosip keycloak-clients 
+5. Updated domain name admin-ui, prereg-ui configmaps. 
+6. Restart mosip modules.
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:9]: Restart MOSIP modules.

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+# Change domain name in a running env ( v3 )
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:11]: Change domain name in a running environment (V3)

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+     ```
+     Example:
+     ```
+     sudo certbot certonly --agree-tos --manual --preferred-challenges=dns -d *.data.mosip.net -d data.mosip.net
+     ```
+3. Update ssl certs location and domain names in nginx config file `/etc/nginx/nginx.conf` and restart nginx service.
+   ```
+   ssl_certificate /etc/letsencrypt/live/data.mosip.net/fullchain.pem;   
+   ssl_certificate_key /etc/letsencrypt/live/data.mosip.net/privkey.pem;
+   ```
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:32]: Can we have a line space here

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+     listen 216.48.176.35:443 ssl;
+     server_name  api.data.mosip.net prereg.data.mosip.net resident.data.mosip.net esignet.data.mosip.net;
+     location / {
+        proxy_pass                      <http://myPublicIngressUpstream;>
+        proxy_http_version              1.1;
+        ....
+        ....
+       }
+   }
+   ```
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:45]: Add line spacing between paragraphs or code blocks as necessary

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+   ```
+   sudo systemctl restart nginx
+   ```
+4. Update the domain in `global` configmaps on the `default` namespace.
+5. Update the iam domain in `keycloak-host` configmaps on `keycloak` namespace.
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:50]: Shouldn't it be IAM? Please check..

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+   done
+   ```
+7. Copy the `keycloak-host` configmaps on all the required namespaces.
+   ```
+   namespaces=$(kubectl get cm -A  | grep  'keycloak-host' | awk '{print $1}' | grep -v 'keycloak' )
+   for ns in $namespaces; do
+      ./copy_cm_func.sh configmap keycloak-host keycloak $ns
+   done
+   ```
+8. Update the domains in gateways  & virtual service via the below commands:
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:66]: Remove extra space after gateways

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+   sed -i 's/country-data/data/g' istio-gateways.yaml
+   kubectl apply -f istio-gateways.yaml
+   ```
+   ```
+   kubectl get vs -A  -o yaml > istio-virtualservice.yaml
+   sed -i 's/country-data/data/g' istio-virtualservice.yaml
+   kubectl apply -f istio-virtualservice.yaml
+   ```
+9. Update the domain names in `landing-page-index` configmaps in the `landing-page` namespace and restart the landing-page service. 
+10. Update the domain name in all ui configmaps (i.e., admin-ui, prereg-ui, pmp ui, resident ui, esignet ui, compliance ui, healthservices UI, etc...).
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:78]: UI nconfigmaps-- needs capitalization

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+   ```
+   ```
+   kubectl get vs -A  -o yaml > istio-virtualservice.yaml
+   sed -i 's/country-data/data/g' istio-virtualservice.yaml
+   kubectl apply -f istio-virtualservice.yaml
+   ```
+9. Update the domain names in `landing-page-index` configmaps in the `landing-page` namespace and restart the landing-page service. 
+10. Update the domain name in all ui configmaps (i.e., admin-ui, prereg-ui, pmp ui, resident ui, esignet ui, compliance ui, healthservices UI, etc...). 
+11. Update the iam domain url for the `Frontend URL` for the `mosip` realm. 
+12. Update the domain in `regclient-index` configmaps & env variables in regclient deployment  for `regclient`.
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:80]: Registration Client deployment- extra space

File: deployment/v3/docs/change-domain-name-in-a-running-env.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+        - name: client_version_env
+          value: 1.2.0.1-B1
+        - name: client_upgrade_server_env
+          value: https://regclient.data.mosip.net
+        - name: healthcheck_url_env
+          value: https://api-internal.data.mosip.net/v1/syncdata/actuator/health
+        - name: host_name_env
+          value: api-internal.data.mosip.net
+    ```
+13. Restart config-server and all other mosip services.
```
Comment (SwatiPeriwal) [inline deployment/v3/docs/change-domain-name-in-a-running-env.md:93]: MOSIP

**Review by SwatiPeriwal — CHANGES_REQUESTED** [review]
Please update as needed

File: deployment/v3/docs/change-domain-name-in-a-running-env.md (hunk 10 above)
Comment (syedsalman3753) [reply deployment/v3/docs/change-domain-name-in-a-running-env.md:50]: It is a keycloak namespace

IAM is only used for keycloak.

**Review by SwatiPeriwal — APPROVED** [review]
Approved chanegs

**Review by ckm007 — APPROVED** [review]

**Pull request merged**